[Regression] Cannot find `System.ValueTuple` when multi-targeting `net462` and `net10.0` using .NET 10.0 RC2

### Describe the bug
When multi-targeting `net462` and `net10.0` using .NET 10.0 RC2, the compiler fails to find type `System.ValueTuple` when building `net462`.

### To Reproduce
Build this with .NET 10 SDK RC2
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>net462;net10.0</TargetFrameworks>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.ValueTuple" Version="4.6.1" />
  </ItemGroup>
</Project>
```
```cs
public class Class1
{
    public (int, int) GetTuple() => (1, 2);
}
```

result:
```
Restore complete (0.5s)
    info NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  test1 net10.0 succeeded (0.1s) → bin\Debug\net10.0\test1.dll
  test1 net462 failed with 1 error(s) (0.1s)
    D:\ws\BugRepos\NET10RC2ValueTuple\test1\Class1.cs(3,12): error CS8179: Predefined type 'System.ValueTuple`2' is not defined or imported

Build failed with 1 error(s) in 1.0s
```

This is a regression compared to RC1.
This doesn't occur with a target framework newer than 462 or older than .NET 10.
This error also doesn't occur when adding a project reference to an empty library targeting `netstandard2.0`.

### Further technical details
<details><summary>details of dotnet --info</summary>

```
.NET SDK:
 Version:           10.0.100-rc.2.25502.107
 Commit:            89c8f6a112
 Workload version:  10.0.100-manifests.4d32cd9e
 MSBuild version:   18.0.0-preview-25502-107+89c8f6a11

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.26100
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\10.0.100-rc.2.25502.107\

.NET workloads installed:
There are no installed workloads to display.
Configured to use workload sets when installing new manifests.
No workload sets are installed. Run "dotnet workload restore" to install a workload set.

Host:
  Version:      10.0.0-rc.2.25502.107
  Architecture: x64
  Commit:       89c8f6a112

.NET SDKs installed:
  8.0.414 [C:\Program Files\dotnet\sdk]
  9.0.306 [C:\Program Files\dotnet\sdk]
  10.0.100-rc.1.25451.107 [C:\Program Files\dotnet\sdk]
  10.0.100-rc.2.25502.107 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 10.0.0-rc.2.25502.107 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 10.0.0-rc.2.25502.107 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.21 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 9.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 10.0.0-rc.2.25502.107 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]
```

</details>

--- a/src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs
@@ -235,12 +235,6 @@ namespace Microsoft.NET.Build.Tasks
         {
             var nugetFramework = new NuGetFramework(targetFrameworkIdentifier, Version.Parse(targetFrameworkVersion));
 
-            //  FrameworkPackages just has data for .NET Framework 4.6.1, so turn on fallback for anything greater than that so it will resolve to the .NET Framework 4.6.1 data
-            if (!acceptNearestMatch && nugetFramework.IsDesktop() && nugetFramework.Version > new Version(4,6,1))
-            {
-                acceptNearestMatch = true;
-            }
-
             var frameworkPackages = FrameworkPackages.GetFrameworkPackages(nugetFramework, [frameworkReference], acceptNearestMatch)
                 .SelectMany(packages => packages)
                 .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithoutTransitiveProjectRefs.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithoutTransitiveProjectRefs.cs
@@ -82,6 +82,7 @@ namespace Microsoft.NET.Build.Tests
                 "1.exe",
                 "1.pdb",
                 "1.exe.config",
+                "System.Diagnostics.DiagnosticSource.dll",
             };
 
             foreach (var targetFramework in targetFrameworks)

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToResolveConflicts.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToResolveConflicts.cs
@@ -329,8 +329,8 @@ namespace Microsoft.NET.Build.Tests
         [InlineData("netstandard1.1", false)]
         [InlineData("netstandard1.0", false)]
         [InlineData("net451", false)]
-        [InlineData("net462")]
-        [InlineData("net481")]
+        [InlineData("net462", false)]
+        [InlineData("net481", false)]
         //  These target frameworks shouldn't prune packages unless explicitly enabled
         [InlineData("net9.0", false, "")]
         [InlineData("netstandard2.1", false, "")]

--- a/test/Microsoft.NET.Build.Tests/GivenThatWeWantToResolveConflicts.cs
+++ b/test/Microsoft.NET.Build.Tests/GivenThatWeWantToResolveConflicts.cs
@@ -492,6 +492,45 @@ namespace Microsoft.NET.Build.Tests
             }
         }
 
+        [Fact]
+        public void WithMultitargetedProject_NETFrameworkIsNotPruned()
+        {
+            var project = new TestProject("MultitargetedPruning")
+            {
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework + ";net462",
+            };
+            project.PackageReferences.Add(new TestPackageReference("System.ValueTuple", "4.6.1"));
+            project.SourceFiles.Add("Test.cs", @"
+public class Class1
+{
+    public (int, int) GetTuple() => (1, 2);
+}
+");
+            var testAsset = _testAssetsManager.CreateTestProject(project, identifier: "NETFrameworkIsNotPruned");
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+            var assetsFilePath = Path.Combine(buildCommand.GetBaseIntermediateDirectory().FullName, "project.assets.json");
+            var lockFile = LockFileUtilities.GetLockFile(assetsFilePath, new NullLogger());
+
+            foreach (var lockFileTarget in lockFile.Targets)
+            {
+                var valueTupleLibrary = lockFileTarget.Libraries.Where(library => library.Name.Equals("System.ValueTuple", StringComparison.OrdinalIgnoreCase)).Single();
+                var runtimeAssemblies = valueTupleLibrary.RuntimeAssemblies.Where(a => !Path.GetFileName(a.Path).Equals("_._"));
+                var compileTimeAssemblies = valueTupleLibrary.CompileTimeAssemblies.Where(a => !Path.GetFileName(a.Path).Equals("_._"));
+
+                if (lockFileTarget.TargetFramework.Framework.Equals(".NETFramework", StringComparison.OrdinalIgnoreCase))
+                {
+                    runtimeAssemblies.Should().NotBeEmpty();
+                    compileTimeAssemblies.Should().NotBeEmpty();
+                }
+                else
+                {
+                    runtimeAssemblies.Should().BeEmpty();
+                    compileTimeAssemblies.Should().BeEmpty();
+                }
+            }
+        }
+
         static List<KeyValuePair<string, string>> ParsePrunePackageReferenceJson(string json)
         {
             List<KeyValuePair<string, string>> ret = new();

